graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,331 @@
+            raise ValueError(msg)
+
+        # get vertex reordering for Ax
+        # in the form (seedsA, voi, rest in order)
+        nsx1 = np.setdiff1d(np.arange(A.shape[0]), np.append(seedsA, voi))
+        a_reord = np.append(np.append(seedsA, voi), nsx1)
+
+        # get reordering for B in the form (seedsB, rest in numerical order)
+        nsx2 = np.setdiff1d(np.arange(B.shape[0]), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        # Reorder the two graphs with our new vertices order
+        AA = A[a_reord][:, a_reord]
+        BB = B[b_reord][:, b_reord]
+
+        # Record where the new seeds and voi locations are
+        # in our re-ordered graphs
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+
+        # Determine what seeds are within a specified subgraph
+        # given by `self.order_voi_subgraph`. If there are no
+        # seeds in this subgraph, print a message and return None
+        subgraph_AA = np.array(
+            _get_induced_subgraph_list(
+                AA, self.order_voi_subgraph, voi_reord, mindist=1
+            )
+        )
+        close_seeds = np.intersect1d(subgraph_AA, seeds_reord)
+
+        if len(close_seeds) <= 0:
+            print(
+                "Voi {} was not a member of the induced subgraph A[{}]".format(
+                    voi, seedsA
+                )
+            )
+            return None
+
+        # Generate the two induced subgraphs that will be used by the matching
+        # algorithm using the seeds that we identified in the previous step.
+        verts_A = np.array(
+            _get_induced_subgraph_list(
+                AA, self.order_seeds_subgraph, list(close_seeds), mindist=0
+            )
+        )
+        verts_B = np.array(
+            _get_induced_subgraph_list(
+                BB, self.order_seeds_subgraph, list(close_seeds), mindist=0
+            )
+        )
+
+        # Determine the final reordering for the graphs that include only
+        # the vertices found by the induced subgraphs in the previous step
+        # For graph A, its of the form (close_seeds, voi, rest in verts_A
+        # in num order). For graph B its of the form (close_seeds, rest in
+        # verts_B in num order)
+        foo = np.append(close_seeds, voi_reord)
+        ind1 = np.append(np.append(close_seeds, voi_reord), np.setdiff1d(verts_A, foo))
+        ind2 = np.concatenate((close_seeds, np.setdiff1d(verts_B, close_seeds)))
+
+        # Generate adjacency matrices for the ordering found in the prev step
+        AA_fin = AA[ind1][:, ind1]
+        BB_fin = BB[ind2][:, ind2]
+
+        # Call the SGM algorithm using random initialization and naive padding
+        # Run the alg on the adjacency matrices we found in the prev step
+        seeds_fin = list(range(len(close_seeds)))
+        sgm = GMP(n_init=self.n_init, shuffle_input=False, init="rand", padding="naive")
```

Comment:
why shuffle_input = False?
Also, I feel like all of the GMP parameters should also be parameters for this class. @bdpedigo thoughts?

Reply:
I chose shuffle_input = False because the code the paper used didn't do this. It seems like a low probability that this algorithm will run into graphs that are already matched. However, I am ok with changing this if that's what y'all want @bdpedigo @asaadeldin11

Reply:
For now I added the user ability to change init, eps, max_iter.

Reply:
turns out the shuffling matters a decent amount due to how linear assignment problem in scipy settles ties... this was a long story we looked into over the summer but it does matter empirically. i trust whatever Ali recommends. 

> Also, I feel like all of the GMP parameters should also be parameters for this class. @bdpedigo thoughts?

yeah i think this makes sense, if you mean a `graph_match_kws={}` kwarg.